Makefile.am

```diff
@@ -224,6 +224,23 @@ clean-local::
 maintainer-clean-local::
 	rm -rf dist/ node_modules/ package-lock.json
 
+BUILT_SOURCES += package-lock.json
+
+package-lock.json: package.json
+	@echo ''
+	@echo '    package-lock.json needs to be (re)generated'
+	@echo ''
+	@echo '    Please run:'
+	@echo ''
+	@echo '        make npm'
+	@echo ''
+	@false
+
+.PHONY: npm
+npm:
+	npm install
```

Comment:
I'm not totally happy about this, though. autogen.sh also does that, with a retry loop, and `npm prune`.  Plus the npm version check (although that may be obsolete now). How about one of these:

 * move the entire logic here, and change autogen.sh to call `make npm`
 * change the message to call autogen.sh
 * move the npm bits into tools/npm-install.sh and calling that from both here and autogen.sh

Reply:
I like the `tools/` approach.  I'll do that.